app/src/main/java/de/tum/in/tumcampusapp/utils/DateTimeUtils.kt

```diff
@@ -0,0 +1,148 @@
+package de.tum.`in`.tumcampusapp.utils
+
+import android.content.Context
+import android.text.format.DateUtils.*
+import de.tum.`in`.tumcampusapp.R
+import org.joda.time.DateTime
+import org.joda.time.format.DateTimeFormat
+import org.joda.time.format.DateTimeFormatter
+import java.text.ParseException
+import java.util.*
+
+object DateTimeUtils {
+    /**
+     * Format an upcoming string nicely by being more precise as time comes closer
+     */
+    fun getFutureTime(time: DateTime, context: Context): String {
+        val timeInMillis = time.millis
+        val now = Calendar.getInstance().timeInMillis
+
+        //Catch future dates: current clock might be running behind
+        if (timeInMillis < now || timeInMillis <= 0) {
+            return DateTimeUtils.getTimeOrDay(time, context)
+        }
+
+        val diff = timeInMillis - now
+        return when {
+            diff < 60 * MINUTE_IN_MILLIS -> {
+                val formatter = DateTimeFormat.forPattern("m")
+                        .withLocale(Locale.ENGLISH)
+                context.getString(R.string.IN) + ' '.toString() + formatter.print(DateTime(diff)) + ' '.toString() + context.getString(R.string.MINUTES)
+            }
+            diff < 3 * HOUR_IN_MILLIS -> { // Be more precise by telling the user the exact time if below 3 hours
+                val formatter = DateTimeFormat.forPattern("HH:mm")
+                        .withLocale(Locale.ENGLISH)
+                context.getString(R.string.AT) + ' '.toString() + formatter.print(time)
+            }
+            else -> getRelativeTimeSpanString(timeInMillis, now, android.text.format.DateUtils.MINUTE_IN_MILLIS, android.text.format.DateUtils.FORMAT_ABBREV_RELATIVE)
+                    .toString()
+        }
+    }
+
+    /**
+     * Format a past timestamp with degrading granularity
+     */
+    fun getTimeOrDayISO(datetime: String, context: Context): String {
```

Comment:
The name of the method is not very descriptive. It’d be great if every method had a more descriptive name and examples in the comment (e.g. what format does it take in and what format does it produce).